explorer/scan/local_scanner.go

```diff
@@ -194,6 +194,9 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstreamConf
 	progressBarElements := map[string]string{}
 	orderedKeys := []string{}
 	for i := range assetList {
+		if assetList[i] == nil || len(assetList[i].PlatformIds) == 0 {
```

Comment:
I would argue this needs to happen in the `im.GetAssets()` function. Having nil, also hints at a problem in the discovery code. Therefore we may need to add additional error handling when assets are added to the inventory.

Reply:
I agree with this. It seems strange to me that an asset can be `nil` or that it can have no platform ID. It seems like we are trying to "fix" discovery bugs with this code